es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
Factoring out the `typeof x` might actually lead to a slowdown.  `typeof x == <string constant>` is optimized as a complete expression, breaking it up might defeat the optimization.  I'll check that.  (There are no actual string comparisons performed after the optimization.)